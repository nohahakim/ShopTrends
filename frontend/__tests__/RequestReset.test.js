import { render, screen } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import userEvent from "@testing-library/user-event";
import RequestReset, {
  REQUEST_RESET_MUTATION,
} from "../components/RequestReset";

const email = "sara@example.com";
const mocks = [
  {
    request: {
      query: REQUEST_RESET_MUTATION,
      variables: { email },
    },
    result: { data: { sendUserPasswordResetLink: null } },
  },
];

describe("<RequestRest />", () => {
  it("renders without crashing", () => {
    const { container } = render(
      <MockedProvider>
        <RequestReset mocks={mocks} />
      </MockedProvider>
    );
    expect(container).toMatchSnapshot();
  });
  it("allows the user to type an email", async () => {
    render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <RequestReset />
      </MockedProvider>
    );
    const emailInput = screen.getByPlaceholderText(/your email/i);
    const signUpButton = screen.getByText(/request reset/i);
    // Simulate typing here
    await userEvent.type(emailInput, email);
    expect(emailInput).toHaveValue(email);
    await userEvent.click(signUpButton);
    const successMessage = await screen.findByText(
      `Success! Check your email for a link!`
    );
    expect(successMessage).toBeInTheDocument();
  });
});
