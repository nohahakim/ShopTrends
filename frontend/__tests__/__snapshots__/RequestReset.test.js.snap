// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestRest /> renders without crashing 1`] = `
<div>
  <form
    class="Form-sc-19c0hb6-0 loFQCy"
    method="POST"
  >
    <h2>
      Request a Password Reset
    </h2>
    <fieldset>
      <label
        for="email"
      >
        Email
        <input
          autocomplete="email"
          name="email"
          placeholder="Your Email Address"
          type="email"
          value=""
        />
      </label>
      <button
        type="submit"
      >
        Request Reset!
      </button>
    </fieldset>
  </form>
</div>
`;
